### PR TITLE
Add support update embed relation (many relationship) with simple object

### PR DESCRIPTION
**Pull request by tuanphpvn**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

*Please update this template with something that matches your PR*

### CODE REVIEW

File: src/Metadata/Property/PropertyNameCollection.php [hunk 1]
```diff
@@ -48,4 +48,12 @@ public function count()
     {
         return count($this->properties);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getArrProperty()
```
Comment (soyuka) [inline src/Metadata/Property/PropertyNameCollection.php]: what does this mean?

Comment (tuanphpvn) [reply src/Metadata/Property/PropertyNameCollection.php]: Never mind. Removed.

**Comment by tuanphpvn** [comment]
Just want to support an simple case:

For example, Update person entity.

with request data change from

```
{
  "name": "Update person name",
  "books": [{
    "id": "existing id"
    "name": "new book name"
   }
  ]
}
```

to

```
{
  "name": "Update person name",
  "books": {
    "id": "existing id"
    "name": "new book name"
   }
}
```

**Comment by soyuka** [comment]
I find this wrong, really not sure that we should accept such request.

**Comment by tuanphpvn** [comment]
Yes, I know. By example above books is array (many relationship). So this implement just give a little more flexible. 

And about definition.

I think: 

1) Array have many item => we should implement array.
2) Array just have one item => we should allow flexible change between array and item. Like

```
books: [{
    "id": "1",
    "name: "name"
}]
```

should equal to

```
books: {
    "id": "1",
    "name: "name"
}
```

If you don't like it. Just close this request. Thank you.

**Comment by soyuka** [comment]
Feel free to add your own normalizer on top of our to handle your custom case. We can not accept this pull request because of the above comments but thank your for your interest in Api Platform.

**Pull request closed without merging**